sklearn/linear_model/ridge.py

```diff
@@ -1270,21 +1271,21 @@ def _eigen_decompose_covariance(self, X, y, sqrt_sw):
             cov[:, -1] = 0
             cov[-1, -1] = sqrt_sw.dot(sqrt_sw)
         nullspace_dim = max(0, X.shape[1] - X.shape[0])
-        s, V = linalg.eigh(cov)
+        lam, Q = linalg.eigh(cov)
         # remove eigenvalues and vectors in the null space of X^T.X
-        s = s[nullspace_dim:]
-        V = V[:, nullspace_dim:]
-        return X_mean, s, V, X
+        lam = lam[nullspace_dim:]
+        Q = Q[:, nullspace_dim:]
```

Comment:
in the reference Q and Lambda are the eigenvectors and eigenvalues of XX^T, whereas V and S^2 are the eigenvectors and eigenvalues of X^TX. I don't think we should be using Q to denote the right singular vectors of X (Q would be closer to U in the notations of section 5; Q[:, :p] = U )